### PR TITLE
Add integration tests for comments

### LINKED ISSUE #22367: Security Check: Comments
- [ ] Missing CSRF checks.
 - [ ] https://github.com/owncloud/core/issues/22368
- [ ] …

### PR DESCRIPTION
**Pull request by LukasReschke**
This adds integration tests for the comments. Especially with regard to the permission handling, time to add automated security integration tests to our APIs.

It's not the super nicest approach but it works :wink: 
Fixes https://github.com/owncloud/core/issues/22367

cc @MorrisJobke @DeepDiver1975 Please review.

### CODE REVIEW

File: build/integration/features/comments.feature [hunk 1]
```diff
@@ -0,0 +1,228 @@
+Feature: comments
+  Scenario: Creating a comment on a file belonging to myself
+	Given user "user0" exists
+    And As an "user0"
+	And User "user0" uploads file "data/textfile.txt" to "/myFileToComment.txt"
+	And "User0" posts a comment with content "My first comment" on the file named "/myFileToComment.txt" it should return "201"
```
Comment (LukasReschke) [inline build/integration/features/comments.feature]: Indentation :speak_no_evil:

File: build/integration/features/comments.feature [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+    And the response should contain a property "oc:parentId" with value "0"
+    And the response should contain a property "oc:childrenCount" with value "0"
+    And the response should contain a property "oc:verb" with value "comment"
+    And the response should contain a property "oc:actorType" with value "users"
+    And the response should contain a property "oc:objectType" with value "files"
+    And the response should contain a property "oc:message" with value "My first comment"
+    And the response should contain a property "oc:actorDisplayName" with value "user1"
+    And the response should contain only "1" comments
+    Then As "user1" delete the created comment it should return "204"
+    Then As "user1 load all the comments of the file named "/myFileToComment.txt" it should return "207"
```
Comment (MorrisJobke) [inline build/integration/features/comments.feature]: missing `"`

File: build/integration/features/bootstrap/Sharing.php [hunk 3]
```diff
@@ -370,5 +370,50 @@ public function checkShareFields($body){
...
+					$user,
+					'123456',
+				],
+				'headers' => [
+					'Content-Type' => 'application/json',
+				],
+			]
+		);
+		$json = json_decode($res->getBody()->getContents(), true);
+		var_dump($res->getBody()->getContents());
```
Comment (LukasReschke) [inline build/integration/features/bootstrap/Sharing.php]: :fire:

File: build/integration/features/comments.feature (hunk 2 above)
Comment (LukasReschke) [inline build/integration/features/comments.feature]: Good catch :smile:

**Comment by SergioBertolinSG** [comment]
I think the base of the gherkin syntax is being forgotten lately, 'given' for preconditions, 'when' for the actual test and 'then' for expected postconditions.

Here the scenarios doesn't have a when statement.

**Comment by LukasReschke** [comment]
@SergioBertolinSG Mind taking another look? THX.

File: build/integration/features/comments.feature [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+    When As "user0" load all the comments of the file named "/myFileToComment.txt" it should return "207"
+    Then the response should contain a property "oc:parentId" with value "0"
+    And the response should contain a property "oc:childrenCount" with value "0"
+    And the response should contain a property "oc:verb" with value "comment"
+    And the response should contain a property "oc:actorType" with value "users"
+    And the response should contain a property "oc:objectType" with value "files"
+    And the response should contain a property "oc:message" with value "My first comment"
+    And the response should contain a property "oc:actorDisplayName" with value "user1"
+    And the response should contain only "1" comments
+    Then As "user0" edit the last created comment and set text to "My edited comment" it should return "403"
```
Comment (rullzer) [inline build/integration/features/comments.feature]: newline ;)

**Comment by rullzer** [comment]
:+1: cool stuff!

File: build/integration/features/comments.feature (hunk 4 above)
Comment (LukasReschke) [inline build/integration/features/comments.feature]: Done :smile:

**Comment by PVince81** [comment]
Nice :+1:

**Comment by blizzz** [comment]
Thank you.

**Pull request merged**